photon-core/src/main/java/org/photonvision/raspi/LibCameraJNILoader.java

```diff
@@ -31,35 +31,47 @@ public class LibCameraJNILoader {
     private static boolean libraryLoaded = false;
     private static final Logger logger = new Logger(LibCameraJNILoader.class, LogGroup.Camera);
 
+    private static File extractLibrary(String libraryName) throws IOException {
+        // We always extract the shared object (we could hash each so, but that's a lot of work)
+        var arch_name = "linuxarm64";
+        var nativeLibName = System.mapLibraryName(libraryName);
+        var resourcePath = "/nativelibraries/" + arch_name + "/" + nativeLibName;
+        var in = LibCameraJNILoader.class.getResourceAsStream(resourcePath);
+
+        if (in == null) {
+            logger.error("Failed to find internal native library at path " + resourcePath);
+            libraryLoaded = false;
+            throw new IOException();
+        }
+
+        // It's important that we don't mangle the names of these files on Windows at least
+        File temp = new File(System.getProperty("java.io.tmpdir"), nativeLibName);
+        FileOutputStream fos = new FileOutputStream(temp);
+
+        int read = -1;
+        byte[] buffer = new byte[1024];
+        while ((read = in.read(buffer)) != -1) {
+            fos.write(buffer, 0, read);
+        }
+        fos.close();
+        in.close();
+
+        return temp;
+    }
+
     public static synchronized void forceLoad() throws IOException {
         if (libraryLoaded) return;
 
         var libraryName = "photonlibcamera";
 
         try {
-            // We always extract the shared object (we could hash each so, but that's a lot of work)
-            var arch_name = "linuxarm64";
-            var nativeLibName = System.mapLibraryName(libraryName);
-            var resourcePath = "/nativelibraries/" + arch_name + "/" + nativeLibName;
-            var in = LibCameraJNILoader.class.getResourceAsStream(resourcePath);
-
-            if (in == null) {
-                logger.error("Failed to find internal native library at path " + resourcePath);
-                libraryLoaded = false;
-                return;
-            }
-
-            // It's important that we don't mangle the names of these files on Windows at least
-            File temp = new File(System.getProperty("java.io.tmpdir"), nativeLibName);
-            FileOutputStream fos = new FileOutputStream(temp);
+            // Development aid. First, try to load the library from disc if it was built locally.
+            File temp = new File("/home/pi/photon-libcamera-gl-driver/cmake_build/libphotonlibcamera.so");
```

Comment:
skull

Reply:
(this is the only merge-blocking comment)

Reply:
Cool. Do we wanna leave some hook around to load a locally-rebuilt .so rather than the packaged one? I might be able to do it with command line arguments. Not a hill I'd die on though, would happily remove for code cleanliness.

Reply:
Unless we see ourselves futzing with libcamera a lot more, let's not add complexity we don't need?